src/wagtail_vector_index/storage/base.py

```diff
@@ -152,6 +164,55 @@ def query(
         response = chat_backend.chat(messages=messages)
         return QueryResponse(response=response.choices[0], sources=sources)
 
+    async def aquery(
+        self, query: str, *, sources_limit: int = 5, chat_backend_alias: str = "default"
+    ) -> AsyncQueryResponse:
+        """
+        Replicates the features of `VectorIndex.query()`, but in an async way.
+        """
+        try:
+            # TODO: use `next()` instead of `[0]` once the `aembed()` method of the
+            #  LiteLLMBackends class is updated to return the same data structure than
+            #  its BaseEmbeddingBackend parent class.
```

Comment:
I remember us mentioning this issue, but I don't recall whether the fix for it was planned or not, @tomusher? :slightly_smiling_face:

Reply:
Not yet, will look in to that soon!